pkg/http/sender.go

```diff
@@ -0,0 +1,638 @@
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) Long256Column(name string, val *big.Int) *LineSender {
+	s.Buffer.Long256Column(name, val)
+	return s
+}
+
+// TimestampColumn adds a timestamp column value to the ILP
+// message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) TimestampColumn(name string, ts time.Time) *LineSender {
+	s.Buffer.TimestampColumn(name, ts)
+	return s
+}
+
+// Float64Column adds a 64-bit float (double) column value to the ILP
+// message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) Float64Column(name string, val float64) *LineSender {
+	s.Buffer.Float64Column(name, val)
+	return s
+}
+
+// StringColumn adds a string column value to the ILP message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) StringColumn(name, val string) *LineSender {
+	s.Buffer.StringColumn(name, val)
+	return s
+}
+
+// BoolColumn adds a boolean column value to the ILP message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) BoolColumn(name string, val bool) *LineSender {
+	s.Buffer.BoolColumn(name, val)
+	return s
+}
+
+// Close closes the underlying HTTP client. If no clients remain open,
+// the global http.Transport will close all idle connections.
+func (s *LineSender) Close() {
+	newCt := clientCt.Add(-1)
+	if newCt == 0 {
+		globalTransport.CloseIdleConnections()
+	}
+}
+
+// AtNow omits the timestamp and finalizes the ILP message.
+// The server will insert each message using the system clock
+// as the row timestamp.
+//
+// If the underlying buffer reaches configured capacity or the
+// number of buffered messages exceeds the auto-flush trigger, this
+// method also sends the accumulated messages.
+func (s *LineSender) AtNow(ctx context.Context) error {
+	return s.At(ctx, time.Time{})
+}
+
+// At sets the timestamp in Epoch nanoseconds and finalizes
+// the ILP message.
+//
+// If the underlying buffer reaches configured capacity or the
+// number of buffered messages exceeds the auto-flush trigger, this
+// method also sends the accumulated messages.
+//
+// If ts.IsZero(), no timestamp is sent to the server.
+func (s *LineSender) At(ctx context.Context, ts time.Time) error {
+	sendTs := true
+	if ts.IsZero() {
+		sendTs = false
+	}
+	err := s.Buffer.At(ts, sendTs)
+	if err != nil {
+		return err
+	}
+
+	s.msgCount++
+
+	if s.autoFlush && s.msgCount > s.autoFlushRows {
```

Comment:
Should be enough to have `s.msgCount == s.autoFlushRows` as the condition here. This way we'll flush when we have exactly `autoFlushRows` rows sitting in the buffer.